src/macaron/slsa_analyzer/git_url.py

```diff
@@ -74,10 +73,12 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "",
     If ``digest`` is provided, this method will checkout that specific commit. If ``digest``
     cannot be found in the current branch, this method will pull (fast-forward only) from remote.
 
-    This method supports repositories which are cloned (full or shallow) from existing remote repositories.
+    This method supports repositories which are cloned from existing remote repositories.
     Other scenarios are not covered (e.g. a newly initiated repository).
 
-    If ``offline_mode`` is set, this method will not pull from remote while checking out the branch or commit.
+    This method will not try to un-shallow an existing shallow-clone repository.
```

Comment:
Not sure why you want to convey here?

Reply:
Hm right, because I haven't decided to remove support for un-shallow repositories, I probably shouldn't mention it too. I will remove this line as it's vague and confusing.

Reply:
Removed in https://github.com/oracle/macaron/pull/491/commits/54ed439f235d5e724550aad4e6c9670b18d2aad7